pkg/tests/apis/helper.go

```diff
@@ -637,3 +638,87 @@ func (c *K8sTestHelper) CreateTeam(name, email string, orgID int64) team.Team {
 	require.NoError(c.t, err)
 	return team
 }
+
+// GenericClient is the struct that implements a generic interface for resource operations
+type GenericClient[T any, L any] struct {
```

Comment:
bubbling this up from a slack thread: @IfSentient mentioned [TypedStore](https://pkg.go.dev/github.com/grafana/grafana-app-sdk@v0.23.1/resource#TypedStore) as an option

Reply:
I had some difficulties migrating to TypedStore. This will require redesigning some integration tests. So, I think for now, for the sake of simplicity, I will stick with GenericClient. I  will rename it to TypedClient to have similar name